### DISCUSSION TITLE
How I can contribute with bugs reports or fixes to the Figma library?

### DISCUSSION THREAD

**Discussion by dfernandez79** [phosphor-icons/homepage#580] [Q&A] [answered]
First of all, thank you to all the people behind Phosphor! The icons are well-crafted and complete.

Recently, I found some "bugs" with the Figma library v2.1. For example, switching the icon color does not always change all the strokes in the icon. 

I started to fix it manually in my library copy, but I guess a script or import plugin generates the Figma version from the core assets.

How can I help by reporting the issues or, if possible, looking for a fix? I would appreciate any pointers. Thank you!

**Comment by rektdeckard** [comment, marked as answer]
Hi @dfernandez79, and thank you!

If you have specific bugs around the Figma library to report you can do that here. For problems with the plugin, you can create issues over at https://github.com/phosphor-icons/figma/issues.

- Which **Format** variant are you using?
- Do specific icons not recolor properly, or does the problem affect all icons of specific variants? 
- How are you changing the color, by using the **Selection colors**, or using the **Fill** or **Stroke** settings directly? 

It's worth pointing out that the **Format** variants are composed very differently, and the correct property to set in order to properly change the color depends on whether you are using the Outline format (which use Fill internally) or the Stroke format (which use Stroke color internally). Usually your best bet is to just use **Selection colors** so you don't have to think about it.

**Reply by dfernandez79** [reply]
Thank you!

I found the cause. While it's related to the **Format** variant, I don't think it is something that Phosphor can fix as it looks more like a Figma bug (or feature?):

Steps to reproduce:
1. Insert an icon component from the Figma library with the Stroke format.
2. Use **Selection colors* to change the icon color.
3. Switch the icon instance to another icon that contains vectors with separate paths. Figma doesn't re-apply the color to all the paths.

It seems like a bug in Figma because Selection Colors works as expected, but the instance swap action fails. This is more problematic when you compose icons in components using instance swap properties. As you pointed out, the solution is to use the Outline variant, as Figma re-applies the fill property correctly. This behavior may be a Figma feature rather than a bug, as it's more conservative about how it reapplies colors on instance swapping.

However, there is another small thing that looks like a Figma bug. While all the components in the Figma library use Outline as the default value for the Format property, when you drag and drop assets from the library, Figma always picks Stroke as the default format. I tried to change it in my library copy without success.
